rhcephcompose/gather/koji.py

```diff
@@ -0,0 +1,72 @@
+from hashlib import md5
+import posixpath
+from rhcephcompose import common_koji
+from rhcephcompose.build import Build
+from rhcephcompose.artifacts import BinaryArtifact, SourceArtifact
+from rhcephcompose.log import log
+
+
+# Koji archive types that represent a Debian source package:
+SOURCE_TYPES = ('tar', 'dsc')
+
+
+def query(profile, tag, whitelist):
+    """
+    Find all the artifacts we will need from a given Koji tag.
+
+    :param profile: Koji profile name, eg "koji"
+    :param tag: Koji tag, "ceph-3.2-xenial"
+    :param whitelist: whitelist of deb package names to save
+    :returns: list of rhcephcompose Build objects
+    """
+    if not common_koji.HAS_KOJI:
+        raise RuntimeError('Please install the koji Python library')
+    # Query koji for all builds in the tag.
+    session = common_koji.get_session(profile)
+    buildinfolist = session.listTagged(tag, inherit=True, latest=True,
+                                       type='debian')
+    pathinfo = common_koji.get_koji_pathinfo(profile)
+    # Query koji for all files in all builds.
+    builds = set()
+    for buildinfo in buildinfolist:
+        build = Build(buildinfo['name'])
+        # Get the URL of the directory for this build.
+        directory = pathinfo.typedir(buildinfo, 'debian')
+        # Look up the list of files for this build.
+        files = session.listArchives(buildinfo['id'])
+        # Find all the .debs (and corresponding dbg .debs) in our whitelist.
+        for filedata in [f for f in files if f['type_name'] == 'deb']:
```

Comment:
this is more of a nit, but I am not used to seeing a loop start with a list comprehension. One way to do this could be something like:

```suggestion
        for filedata in files:
            if filedata['type_name'] != 'deb':
                continue
```

These list comprehensions get gnarly when other conditions need to be added. Again, just a nit, not that you need to update this

Reply:
Great point, and thanks for the patch. I've reformatted this now in this PR.